frontend/src/app/components/faucet/faucet.component.ts

```diff
@@ -135,9 +129,40 @@ export class FaucetComponent implements OnInit, OnDestroy {
     });
   }
 
+  isDisabled(): boolean {
+    return !(this.user && this.status?.code === 'ok' && !this.error);
+  }
+
+  getNotFaucetAddressValidator(faucetAddress: string): ValidatorFn {
+    return faucetAddress ? (control: AbstractControl): ValidationErrors | null => {
+      const forbidden = control.value === faucetAddress;
+      return forbidden ? { forbiddenAddress: { value: control.value } } : null;
+    }: () => null;
+  }
+
+  initForm(min: number, max: number, faucetAddress: string): void {
+    this.faucetForm = this.formBuilder.group({
+      'address': ['', [Validators.required, Validators.pattern(getRegex('address', 'testnet4')), this.getNotFaucetAddressValidator(faucetAddress)]],
+      'satoshis': [min, [Validators.required, Validators.min(min), Validators.max(max)]]
+    });
+
+    this.loading = false;
+    this.cd.markForCheck();
+  }
+
+  updateForm(min, max, faucetAddress: string): void {
```

Comment:
Also set the default values here. EG: We set min to 1000 sats by default, but if status send a min of 5000 sats, you can still attempt a request and it ends up with a form validation error in the backend

```
May 21 10:10:44 [25666] DEBUG: <services> Validation error: The string input 1000 failed the validation function
May 21 10:10:44 [25666] DEBUG: <services> Invalid format for sats
```

Reply:
fixed